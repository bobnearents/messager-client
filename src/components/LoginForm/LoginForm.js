import React, { Component } from "react";
import TokenService from "../../services/token-service";
import AuthApiService from "../../services/auth-api-service";
import MessageContext from "../../context/message-context";
import { Link } from "react-router-dom";
import './LoginForm.css';

class LoginForm extends Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  static contextType = MessageContext;

  handleSubmitJwtAuth = ev => {
    ev.preventDefault();
    this.setState({ error: null });
    this.context.loadingTrue();
    const { username, password } = ev.target;
    AuthApiService.postLogin({
      username: username.value,
      password: password.value
    })
      .then(res => {
        username.value = "";
        password.value = "";
        TokenService.saveAuthToken(res.authToken);
        this.props.onLoginSuccess();
        this.context.loadingFalse();
      })
      .catch(res => {this.setState({error: res.error})})
  };

  render() {

    const error = this.state.error;
    return (
      <div className="signup-form">
        <h2 className='form-header'>Log in!</h2>
        <form onSubmit={ev => this.handleSubmitJwtAuth(ev)}>
          {error && <span className='error'>{error}</span>}
          <label htmlFor="username">Username: </label>
          <input type="text" id="username" />
          <label htmlFor="password">Password: </label>
          <input type="password" id="password" />
          <button type="submit">Submit</button>

          <div>
            Don't have an account?
            <Link to='/signup'>Make one here</Link>
          </div>
          
        </form>
      </div>
    );
  }
}

export default LoginForm;
