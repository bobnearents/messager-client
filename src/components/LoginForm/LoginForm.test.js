import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import { Route, Link, MemoryRouter } from "react-router-dom";

import LoginForm from "./LoginForm";

describe("<LoginForm />", () => {
  it("Renders without crashing", () => {
    shallow(<LoginForm />);
  });

  it("renders correctly", () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <LoginForm />
        </MemoryRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
