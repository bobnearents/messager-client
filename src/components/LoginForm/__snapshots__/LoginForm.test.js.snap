// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginForm /> renders correctly 1`] = `
<div
  className="signup-form"
>
  <h2
    className="form-header"
  >
    Log in!
  </h2>
  <form
    onSubmit={[Function]}
  >
    <label
      htmlFor="username"
    >
      Username: 
    </label>
    <input
      id="username"
      type="text"
    />
    <label
      htmlFor="password"
    >
      Password: 
    </label>
    <input
      id="password"
      type="password"
    />
    <button
      type="submit"
    >
      Submit
    </button>
    <div>
      Don't have an account?
      <a
        href="/signup"
        onClick={[Function]}
      >
        Make one here
      </a>
    </div>
  </form>
</div>
`;
